From b5f7f7f8f7ee5bb104a0ca3262df0e02e176e219 Mon Sep 17 00:00:00 2001
From: Bogdan Togorean <bogdan.togorean@analog.com>
Date: Fri, 23 Jul 2021 09:07:30 +0300
Subject: [PATCH 16/21] adrivers: media: spi: addicmos.c: Add resolution for
 mode 10

Mode 10 require 4096x256x9 subframes so add corresponding resolution. Also disable default writing of calibration data.

Signed-off-by: Bogdan Togorean <bogdan.togorean@analog.com>
---
 drivers/media/spi/addicmos.c | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/drivers/media/spi/addicmos.c b/drivers/media/spi/addicmos.c
index 14faf522d39f..24ce8e03988f 100644
--- a/drivers/media/spi/addicmos.c
+++ b/drivers/media/spi/addicmos.c
@@ -150,6 +150,7 @@ static const struct reg_sequence addicmos_standby_setting[] = {
 };
 
 static const s64 link_freq_tbl[] = {
+	732000000,
 	732000000,
 	732000000,
 	732000000,
@@ -185,9 +186,15 @@ static const struct addicmos_mode_info addicmos_mode_info_data[] = {
 	},
 	{
 		.width = 4096,
-		.height = 2560,
+		.height = 2304,
 		.pixel_rate = 488000000,
 		.link_freq_idx = 4 /* an index in link_freq_tbl[] */
+	},
+	{
+		.width = 4096,
+		.height = 2560,
+		.pixel_rate = 488000000,
+		.link_freq_idx = 5 /* an index in link_freq_tbl[] */
 	}
 };
 
@@ -798,8 +805,8 @@ static int addicmos_firmware_load(struct v4l2_subdev *sd)
 		return ret;
 	}
 
-	//Writes for Default calibration
-#if 1
+	//Writes for Default calibration - disabled
+#if 0
 	regmap_multi_reg_write(addicmos->regmap,
 			addicmos_powerup_setting,
 			ARRAY_SIZE(addicmos_powerup_setting));
-- 
2.17.1

